src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -124,8 +183,11 @@ private void IconBitmapImage_DownloadOrDecodeFailed(object sender, System.Window
         {
             var bitmapImage = sender as BitmapImage;
 
-            // Fix the bitmap image cache to have default package icon, if some other failure didn't already do that.
-            var cachedBitmapImage = _bitmapImageCache.Get(bitmapImage.UriSource.ToString()) as BitmapSource;
+            var uri = bitmapImage.UriSource;
+
+            string cacheKey = uri != null ? uri.ToString() : string.Empty;
```

Comment:
Any possibility of uri being whitespace? Then change != null to a IsNullOrWhitespace check, here.

Reply:
uri cannot be whitespace, since it is an `Uri` type. Uri class don't allow URIs only formed by spaces.